### DISCUSSION TITLE
Triton server model repo does not reflect MLFlow version

### DISCUSSION THREAD

**Discussion by jackroseman** [triton-inference-server/server#5691] [Q&A]
Hello,

I am trying to understand better if there is a requirement for the triton model repository to only have a version "1". What I see is when I use the mlflow-triton plugin, if the model, call it "model1" is versioned as "6" in MLFlow, then I would expect the model repo to look like the following:

```
models
└───model1
│   │   config.pbtxt
│   │   mlflow-meta.json
│   └───6
│       │   model.onnx
```
Looking at the mlflow-triton plugin code, I see that the model repo will replace the model version in this file structure with "1". This is alright because if I deploy the model by the model version number using the `create_deployment(model_name, model_uri, flavor)`, the `mlflow-meta.json` file has the mapping to the mlflow model uri, which can tell me the MLFlow version that was deployed. You see, the issue with this is that I am able to specify the model_uri to be `models:/model1/staging` which does not give me traceability to the mlflow model version if the model moves out of staging for some reason. 

For my use case, I find it both important to not specify the mlflow model version during deployment ('models:/model1/production'). However, once a model is in 'production', it is required for me to ensure full traceability to the mlflow model version.

Is there a requirement on the model repo to always use "1" as the model version in the repository? I am considering either modifying the model repo, or the information in the `mlflow-meta.json` to preserve the version even when I pass in model stage as the model URI.

Thanks

**Comment by the-david-oy** [comment]
CC: @GuanLuo
